genesyscloud/dependent_consumers/genesyscloud_dependent_consumer_proxy.go

```diff
@@ -142,24 +151,49 @@ func buildDependsMap(resources resourceExporter.ResourceIDMetaMap, dependsMap ma
 
 // This private function includes iteration of the dependent Consumers and build DependsList for each Resource
 // This also checks for dependent flows and again export those dependencies
-func iterateDependencies(dependencies *platformclientv2.Consumedresourcesentitylisting, resources resourceExporter.ResourceIDMetaMap, dependsMap map[string][]string, ctx context.Context, p *DependentConsumerProxy, key string) (resourceExporter.ResourceIDMetaMap, map[string][]string, error) {
+func iterateDependencies(dependencies *platformclientv2.Consumedresourcesentitylisting, resources resourceExporter.ResourceIDMetaMap, dependsMap map[string][]string, ctx context.Context, p *DependentConsumerProxy, key string,
+	architectDependencies map[string][]string) (resourceExporter.ResourceIDMetaMap, map[string][]string, error) {
 	dependentConsumerMap := SetDependentObjectMaps()
 	for _, consumer := range *dependencies.Entities {
```

Comment:
This function is getting pretty complex.  We should look at breaking it up into multiple functions just to make it easier to unit test.

Since this is for a bug, I would create a ticket for refactor and then assign it for the next sprint.  If you feel this will fix the code lets get this merged